fdbserver/workloads/AtomicOps.actor.cpp

```diff
@@ -172,29 +181,154 @@ struct AtomicOpsWorkload : TestWorkload {
 			loop {
 				try {
 					int group = deterministicRandom()->randomInt(0,100);
-					uint64_t intValue = deterministicRandom()->randomInt( 0, 10000000 );
+					state uint64_t intValue = deterministicRandom()->randomInt(0, 10000000);
 					Key val = StringRef((const uint8_t*) &intValue, sizeof(intValue));
-					tr.set(self->logKey(group), val);
+					std::pair<Key, Key> logDebugKey = self->logDebugKey(group);
 					int nodeIndex = deterministicRandom()->randomInt(0, self->nodeCount / 100);
-					tr.atomicOp(StringRef(format("ops%08x%08x", group, nodeIndex)), val, self->opType);
-					// TraceEvent(SevDebug, "AtomicOpWorker")
-					//     .detail("LogKey", self->logKey(group))
-					//     .detail("Value", val)
-					//     .detail("ValueInt", intValue);
-					// TraceEvent(SevDebug, "AtomicOpWorker")
-					//     .detail("OpKey", format("ops%08x%08x", group, nodeIndex))
-					//     .detail("Value", val)
-					//     .detail("ValueInt", intValue)
-					//     .detail("AtomicOp", self->opType);
+					Key opsKey(format("ops%08x%08x", group, nodeIndex));
+					tr.set(logDebugKey.first, val); // set log key
+					tr.set(logDebugKey.second, opsKey); // set debug key; one opsKey can have multiple logs key
+					tr.atomicOp(opsKey, val, self->opType);
 					wait( tr.commit() );
+					if (self->opType == MutationRef::AddValue) {
+						self->lbsum += intValue;
+						self->ubsum += intValue;
+					}
 					break;
 				} catch( Error &e ) {
 					wait( tr.onError(e) );
+					if (self->opType == MutationRef::AddValue) {
+						self->ubsum += intValue;
+					}
 				}
 			}
 		}
 	}
 
+	ACTOR Future<Void> dumpLogKV(Database cx, int g) {
+		try {
+			state ReadYourWritesTransaction tr(cx);
+			Key begin(format("log%08x", g));
+			Standalone<RangeResultRef> log = wait(tr.getRange(KeyRangeRef(begin, strinc(begin)), CLIENT_KNOBS->TOO_MANY));
```

Comment:
The potential for confusion with these limits still exists if they are hit. I think we should either prove that the limits can't be hit and `ASSERT` or add a trace event to signal that they have been hit.

Reply:
Will add a trace event.

Reply:
I think you forgot this trace event (and the analogous ones in the following 2 functions).